pydantic/_internal/_generate_schema.py

```diff
@@ -683,7 +683,7 @@ def _generate_parameter_schema(
             field = FieldInfo.from_annotated_attribute(annotation, default)
         assert field.annotation is not None, 'field.annotation should not be None when generating a schema'
         schema = self.generate_schema(field.annotation)
-        schema = apply_annotations(schema, field.metadata, self.definitions)
+        schema = apply_annotations(schema, field.metadata, self.definitions, annotation)
```

Comment:
One fair question when looking at this is "what if `annotation` is an `Annotated`?". It ends up not being a problem and the user always gets the `Annotated` unwrapped because:
1. `Annotated` self flatens: `assert Annotated[Annotated[str, "foo"], "bar"] == Annotated[str, 'foo', 'bar']`
2. We must have just unwrapped an `Annotated` to get here, and because of (1) they can't be immediately nested.

Reply:
That said, if you have something like `Annotated[List[Annotated[int, …], SomeMetadata]` then SomeMetadata should (need to add a test) get a `List[Annotated[…]]`